File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.java
class Solution {
    public String solution(String my_string, int[][] queries) {
        String answer = "";
        for(int i =0; i<queries.length; i++){
            String b = "";
            for(int j=queries[i][1]; j>=queries[i][0]; j--){
                b+= my_string.charAt(j);
            }
             String a = my_string.substring(0,queries[i][0]);
             String c = my_string.substring(queries[i][1]+1);
            answer  = a + b + c;
            }
        return answer;
    }
}

File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.js
function solution(my_string, queries) {
    for (let i = 0; i < queries.length; i++) {
        const [s, e] = queries[i]; // 구조분해할당 - 각 query에서 s와 e 가져오기
        const reversedPart = my_string.slice(s, e + 1).split("").reverse().join(""); 
        my_string = my_string.slice(0, s) + reversedPart + my_string.slice(e + 1); 
    }
    return my_string;
}
